roles/deploy-aws-infra/tasks/main.yml

```diff
@@ -0,0 +1,343 @@
+  ec2_vpc_net:
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    name: "{{ env_id }}-vpc"
+    cidr_block: 172.31.0.0/16
+    region: "{{ aws_region }}"
+    state: "present"
+    tags:
+      env_id: "{{ env_id }}"
+  when: aws_create_vpc
+  register: new_vpc
+
+- name: Create Subnets for the new VPC (HA Mode)
+  ec2_vpc_subnet:
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    cidr: 172.31.{{ item }}.0/20
+    region: "{{ aws_region }}"
+    vpc_id: "{{ new_vpc.vpc.id }}"
+    map_public: yes
+    az: "{{ aws_region }}a"
+    state: "present"
+    tags:
+      env_id: "{{ env_id }}"
+  with_sequence: start=0 end=64 stride=16
+  when: aws_create_vpc and ha_mode
+  register: new_subnets
+
+- name: Create Subnet for the new VPC (Non HA Mode)
+  ec2_vpc_subnet:
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    cidr: 172.31.0.0/20
+    region: "{{ aws_region }}"
+    vpc_id: "{{ new_vpc.vpc.id }}"
+    map_public: yes
+    az: "{{ aws_region }}a"
+    state: "present"
+    tags:
+      env_id: "{{ env_id }}"
+  when: aws_create_vpc and not ha_mode
+  register: new_subnets
+
+- name: Create SSH Security Group for the new VPC
+  ec2_group:
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    name: "labs-ssh"
+    description: "Labs SSH"
+    vpc_id: "{{ new_vpc.vpc.id }}"
+    region: "{{ aws_region }}"
+    state: "present"
+    tags:
+      env_id: "{{ env_id }}"
+    rules:
+      - proto: tcp
+        from_port: 22
+        to_port: 22
+        cidr_ip: 0.0.0.0/0
+  when: aws_create_vpc
+  register: ssh_sg
+
+- name: Create OCP Master Security Group for the new VPC
+  ec2_group:
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    name: "labs-ocp-master"
+    description: "Labs OCP Master"
+    vpc_id: "{{ new_vpc.vpc.id }}"
+    region: "{{ aws_region }}"
+    state: "present"
+    tags:
+      env_id: "{{ env_id }}"
+    rules:
+      - proto: udp
+        from_port: 8053
+        to_port: 8053
+        cidr_ip: 0.0.0.0/0
+      - proto: tcp
+        from_port: 8053
+        to_port: 8053
+        cidr_ip: 0.0.0.0/0
+      - proto: tcp
+        from_port: 8443
+        to_port: 8443
+        cidr_ip: 0.0.0.0/0
+      - proto: tcp
+        from_port: 443
+        to_port: 443
+        cidr_ip: 0.0.0.0/0
+      - proto: udp
+        from_port: 53
+        to_port: 53
+        cidr_ip: 0.0.0.0/0
+      - proto: tcp
+        from_port: 53
+        to_port: 53
+        cidr_ip: 0.0.0.0/0
+      - proto: udp
+        from_port: 4789
+        to_port: 4789
+        cidr_ip: 0.0.0.0/0
+  when: aws_create_vpc
+  register: ocp_master_sg
+
+- name: Create OCP Node Security Group for the new VPC
+  ec2_group:
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    name: "labs-ocp-node"
```

Comment:
no hardcoded values related to Labs (or CASL)

Reply:
ok